### ISSUE TITLE
[Bug]: Error occurs when trying to unzip the file - won't allow me to remove it either.

### ISSUE THREAD

**Issue by Megs108** [ollm/OpenComic#224]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/ollm/OpenComic/issues) for a bug report that matches the one I want to file, without success.

### OpenComic Version

1.0.0

### Operating System

macOS Catalina 10.15.7

### Steps to reproduce

To whom it may concern,

1. Went to upload a file from my desktop not from documents.
2. This then caused an error ( I was unaware that it would, have know learnt my mistake).
3. I will include a screenshot so you can see the error that gets shown.
![Screenshot 2024-04-28 at 19 36 41](https://github.com/ollm/OpenComic/assets/154489873/2c6d8ef2-9d27-4ebe-ab78-d7a31e69617c)
4.Please help me resolve this issue. I'll explain more below.

### Expected Behavior

I assumed that I would, like usual,  be able to open and view the file like all the others.
 and to be able to remove it from the library.

### Actual Behavior

However this error message kept on showing regardless of whatever action I made on OpenComic. I tried to remove the Desktop file but it won't - just shows the error alert. I tried to reload and force reload but again the error alert. I also cleared the cache which it does do however the error message still shows. With any click the alert shows: go back into the library - shows the same alert, choose a new comic, shows the same alert. So it allows me to move between, eg, library and settings but always shows the alert and the Desktop folder cannot/won't be removed from the library.

### Error message

An error occurred while trying to unzip the file.
EPERM: operation not permitted, scandir '/Users/meg/Desktop'

![Screenshot 2024-04-28 at 19 36 41](https://github.com/ollm/OpenComic/assets/154489873/1bb21db4-5e5e-4fa5-a667-d2b076df9449)

### Additional Information

Also, I do not know where to see what version of your programme I have. I believe it to be the latest as I remember the message that popped up and I clicked on download - this then sent me to the GitHub page - however I was confused and did not know if there was a further step or if it had been upgraded.....? Please inform me on what to do here also for future upgrades.
Thank you.

Regards,
Meg.

**Comment by Megs108** [comment]
Hi, 

It's Megs108 again.
I thought of this after but would the issue be resolved by my deleting/removing OpenComic from my laptop and then re-downloading it from the App Store? 

Please let me know! Thanks.

**Comment by ollm** [comment]
Hi.

If you have downloaded OpenComic from the App Store, the following steps should solve this problem.

Close the error message by clicking Ok, then click on `OpenComic` in the top bar > `Open folder` and open your desktop folder (Or the folder that is giving you an error at this moment), this should solve it.

This is because applications from the App Store do not have reading permissions to read certain folders due the [App Sandbox](https://developer.apple.com/documentation/security/app_sandbox/) and to do this the user has to grant permissions (In this case, the permission is granted by the user through the open folder dialog).

How did you add the folder to OpenComic? Through drag and drop? This should be disabled in the latest version of OpenComic (v1.2.0), as permissions are not granted using this method https://github.com/electron/electron/issues/40678. From the screenshot, it looks like you are using v1.0.0. I recommend that you also update to the latest version through the App Store or from the [Realeses](https://github.com/ollm/OpenComic/releases) page (Depending on where you have it installed from).

Let me know if this can't resolve the error.

**Comment by Megs108** [comment]
Hi,

Thanks for telling me. It worked to remove the file and then I also upgraded the App however now when I open it all it shows is a blank page with just the three buttons top left (close, expand and minimise)..... what should I do now?

**Comment by Megs108** [comment]
Hi,
There is no error message upon opening and nothing happens even after I click on Toggle DevTools. I also tried the shortcut and that didn't work either. Would restarting my computer help this? Or is there something else I can try?

I think - if I understood your comment above - the error might happen later? As I find Toggle under Debug not View. 

![Screenshot 2024-04-29 at 16 27 24](https://github.com/ollm/OpenComic/assets/154489873/c3ce7437-2af2-4f78-a08c-cff0f7a02e0f)

**Comment by ollm** [comment]
> Would restarting my computer help this?

If you haven't tried it yet since the error appeared, you can try it, also try opening again the DevTools after rebooting (Check that it is not opening in a new window)

> I think - if I understood your comment above - the error might happen later? As I find Toggle under Debug not View.

OpenComic modifies the default Electron menu bar (The one in the first image) when it opens, if there is an error before doing so, the Electron bar will be maintained and not the OpenComic bar. Seeing that the OpenComic menu bar does appear, there must be an error between this and when the app is displayed.

> Or is there something else I can try?

You can try to uninstall and reinstall OpenComic again.

From where you installed it? App Store? I can try to push an update to see if the problem is fixed in the master branch.

**Comment by Megs108** [comment]
Hi,
There is no change even after restarting the laptop (checked for a new window and there wasn't any), so I will now uninstall and reinstall the app.
Will let you know how it goes.
Thanking God it works.

**Comment by Megs108** [comment]
Hi,
So I deleted OpenComic from launchpad and then proceeded to open it from the App Store (which is where I got it from originally) however it does not open. I have tried a number of times now and waited in between - incase it was just lagging - but there is no response.
What happens now?

**Comment by ollm** [comment]
You can also try deleting the stored data of the app, in case there are any errors in them.

To do this you have to access one of the following folders and delete all the `.json` files in them (First close the app), the files should have names like `config.json`, `comics.json`, `cache.json`, etc.

`/Users/<youUSerName>/Library/Containers/org.opencomic.app/Data/Library/Application Support/OpenComic/storage` 
`/Users/<youUSerName>/Library/Containers/OpenComic/Data/Library/Application Support/OpenComic/storage` 

If this doesn't fix the issue, I'll push an update to the App Store with some changes that might fix it.

**Comment by Megs108** [comment]
Hi,

Unfortunately I still can't get the application to open from the App Store.

**Comment by ollm** [comment]
Try this build for macOS, is updated to the master branch (First you have to uninstall the App Store version)

PKG: https://mega.nz/file/3b4nQDJT#eoXC6tDybaYu9P9ZBkEfelm8zMK9TvHnyhQubrPVnsY
DMG: https://mega.nz/file/LDZiUa5Q#fPAH6c_SspKW4VPx_ONwr3ppC2B39QgGnoF9w0Lf0EY

If that doesn't solve it, I will soon try to disable some functions that could cause the issue and send you other build.

**Comment by Megs108** [comment]
Hi,

I tried but it comes up with this (below) after I double click the little box in downloads;

![Screenshot 2024-04-30 at 14 19 14](https://github.com/ollm/OpenComic/assets/154489873/9fe1f233-f6d9-49f6-a802-30157006311c)

Is there something I need to do from my end? And is this download safe?

**Comment by ollm** [comment]
> Is there something I need to do from my end? And is this download safe?

Yes, the download is safe, I builded it myself, but by default Apple blocks all software that is not distributed through the App Store or that is correctly notarized ([Notarizing macOS software before distribution](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution)), currently, I am not currently certifying OpenComic through notarization to distribute outside App Store without this problem.

You can force the installation of the app using the following steps:

From: https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/mac

> - In the Finder  on your Mac, locate the app you want to open.
> - Don’t use Launchpad to do this. Launchpad doesn’t allow you to access the shortcut menu.
> - Control-click the app icon, then choose Open from the shortcut menu.
> - Click Open.
> 
> The app is saved as an exception to your security settings, and you can open it in the future by double-clicking it just as you can any registered app.

![Captura desde 2024-04-30 18-54-13](https://github.com/ollm/OpenComic/assets/18549191/0f47903f-26f3-462e-8315-8524af2658e1) ![Captura desde 2024-04-30 18-54-29](https://github.com/ollm/OpenComic/assets/18549191/b13a9a8f-87b0-435e-82a6-89ce1a3d43ae)

**Comment by Megs108** [comment]
Hi,

Thank you for showing me how to open your app with this method. 
Unfortunately the same thing is happening as it did from the App Store download. I can get it to show in launchpad, put it onto the dock and "open" the application but it's still just a blank page.

I tried to do the debug - toggle, but nothing happened.

In this previous message (copied and pasted below) from you, what folders do I need to access (where are they/how do I find them?) Should I even try this or just wait for you to update the application?

You can also try deleting the stored data of the app, in case there are any errors in them.

To do this you have to access one of the following folders and delete all the .json files in them (First close the app), the files should have names like config.json, comics.json, cache.json, etc.

/Users/<youUserName>/Library/Containers/org.opencomic.app/Data/Library/Application Support/OpenComic/storage
/Users/<youUserName>/Library/Containers/OpenComic/Data/Library/Application Support/OpenComic/storage

If this doesn't fix the issue, I'll push an update to the App Store with some changes that might fix it.

**Comment by ollm** [comment]
> PS. Not sure if this will mean anything, but when I have the application open in this state of 'being blank' it requires me to click twice on - for eg - the red x to close it and the same thing when I went to quit it I had to do this twice when usually you just press once and the page would close, quit etc. Again not sure if this is useful to figuring out the issue but I thought the more information I could give the better. Sorry for all the hassle and thanks for working to fix this problem.

This is because OpenComic executes some functions before closing (Save reading progress, delete temporary files and delete old cache (Thumbnails)) but due to the error it does not execute correctly, the second time you click it closes directly without doing all this.

The strange thing is that if there is an error an error message should appear or you should be able to open the DevTool, the only thing I can think of that could cause this is that the app is stuck in a loop. You can try check this with the activity monitor, if OpenComic is using a lot of CPU when it is open in the white screen it is probably stuck in a loop.

![Captura desde 2024-05-01 10-12-58](https://github.com/ollm/OpenComic/assets/18549191/03df3e6c-593a-44c3-809a-6740b0ba06aa)

> In this previous message (copied and pasted below) from you, what folders do I need to access (where are they/how do I find them?) Should I even try this or just wait for you to update the application?

I recommend that you do it, since it could be that some data had not been migrated correctly during the update. I leave you the steps on how to do it.

Copy the following path (already includes your username).

`/Users/meg/Library/Containers/org.opencomic.app/Data/Library/Application Support/OpenComic/storage`

At first this should work, but if not, try the next one.

`/Users/meg/Library/Containers/OpenComic/Data/Library/Application Support/OpenComic/storage`

 In Finder, in the menu bar tap Go > Go to folder.

![Captura desde 2024-05-01 10-01-24](https://github.com/ollm/OpenComic/assets/18549191/0b561b7b-a988-411b-8850-a39d4b1cab15)

Paste the path and press enter.

![Captura desde 2024-05-01 10-08-59](https://github.com/ollm/OpenComic/assets/18549191/1d00ad84-8670-4f16-b33e-2d190d62af66)

The following files should appear, if so delete them all.

![Captura desde 2024-05-01 10-09-15](https://github.com/ollm/OpenComic/assets/18549191/7da3b63c-d30d-48b4-9606-243ca12fed44)

If the error was related to incorrect data, now opening OpenComic should do so without problems.

Even so, I am going to prepare a build by disabling some functions that could be causing the problem, since in the last version I introduced cache data compression 3ae7cb721fb964fa1a41f8a5170b775d2182e8a9 so that they took up less space, and I don't know if this may be causing problems on certain devices.

**Comment by Megs108** [comment]
Hi,

So I checked the activity monitor first but OpenComic CPU stays at 0,0 regardless of how long the application is open for.
I then tried to find the folders (using both paths in the end) however I don't think there is any stored data as I just kept getting shown the following folders you see in the screenshot.

![Screenshot 2024-05-01 at 10 57 46](https://github.com/ollm/OpenComic/assets/154489873/63f7754c-7c03-4efa-878f-e7f585ba4db6)

The list is slightly longer but there was nothing with OpenComic in it's name/title.

**Comment by ollm** [comment]
> I then tried to find the folders (using both paths in the end) however I don't think there is any stored data as I just kept getting shown the following folders you see in the screenshot.

Sry, it seems the location is different if installing via `pkg`, the location above is for the version installed from the App Store, the one installed through `pkg` is the following.

`/Users/meg/Library/Application Support/OpenComic/storage`

Although now I doubt that this is the problem, since the version installed through `pkg` should have use a new data, but you can try it anyway.

-----------

I have prepared some builds with some changes, you should try them in order.

1. This build has app cache compression disabled, the compression used is `zstd` and I think this could be the most likely cause of the problem.
 [OpenComic-without-cache-compression](https://mega.nz/file/nOIG1RBS#0sp2SGUYDUceeHhuUvM64IEHNwX521m0BuFwr02K7FY)

2. This also has cache compression disabled, but uses the version of `electron` that OpenComic v1.0.0 used, `electron` is a framework for creating desktop applications, which is what I use for OpenComic.
 If the previous version has already worked correctly, you can skip directly to the 3rd build.
 [OpenComic-electron-25-without-cache-compression](https://mega.nz/file/OGxV3IKS#QJA4nAI5qb9Kzh0AJDSq3fIR_2pfsggbdnBH3VNDOBs)

3. This does not have app cache compression disabled, but it tries to detect if it fails and will not use it in that case db00a0067ce4b8786b813af96795b37ca970f3db
 You don't need to try it if none of the above has worked for you.
 [OpenComic-error-detection-in-cache-compression-electron-25](https://mega.nz/file/jeg13KaS#2hR-bAQlPT8RnVOFPZKxJQgtG_qxUQMQ2S6cM9n-5eQ)

Another option I would try if everything else doesn't work is to install OpenComic v1.0.0 through the `pkg`, this worked correctly when you had it installed from the App Store, so it is quite likely that it will work correctly.

[OpenComic v1.0.0 PKG from GitHub](https://github.com/ollm/OpenComic/releases/download/v1.0.0/OpenComic-1.0.0.pkg)

**Comment by Megs108** [comment]
Okay I'll give them a go. 
Do I need to first delete the OpenComic I downloaded from the previous package before trying any of the new ones from the list above?

**Comment by ollm** [comment]
It is not necessary since the previous version is replaced, but if you install the v1.0.0 it would be recommended due it would be downgrading.

**Comment by Megs108** [comment]
Okay so the first two did not work but before trying v1.0.0 do I uninstall this version of OpenComic by moving it to bin and deleting from there?

**Comment by ollm** [comment]
Yes, that should be enough, although in my case, having the app installed from different sources, I have had to do it from different locations.

If after uninstalling any of them still appear in the search, you can locate and uninstall them with the following.

![Captura desde 2024-05-02 10-20-09](https://github.com/ollm/OpenComic/assets/18549191/78a576e7-c234-4de9-be55-904183d2379c)

Open the app, in the dock right click `Options` > `Show in Finder`, now close the app and move the one that appeared in Finder to the trash, empty the trash.

![Captura desde 2024-05-02 10-05-51](https://github.com/ollm/OpenComic/assets/18549191/0e39820e-ca11-4a21-8eec-464e32bd4662)

-------------

Now, I think it would be better if you installed version v1.0.0 using the `dmg`, I tried it first with the `pkg`, but it did not install correctly and with the `dmg` it has not given me any problems.

[OpenComic v1.0.0 DMG from GitHub](https://github.com/ollm/OpenComic/releases/download/v1.0.0/OpenComic-1.0.0.dmg)

Installation using `dmg` is done like this:

Double click the `dmg` file, the following window should appear, drag the OpenComic icon to the Applications folder.

![Captura desde 2024-05-02 10-22-39](https://github.com/ollm/OpenComic/assets/18549191/c1257e4f-fc8d-41bc-bd3c-35cced289d5b)

Now open the Applications folder in Finder, I recommend that before opening the app you check that it is v1.0.0, to do so `Right click` > `Get Info`

![Captura desde 2024-05-02 10-17-59](https://github.com/ollm/OpenComic/assets/18549191/534ac8b1-4a93-41b8-8037-688458d9a780) ![Captura desde 2024-05-02 10-19-09](https://github.com/ollm/OpenComic/assets/18549191/45283a78-1814-4b17-ba3a-940697ca23dc)

If everything is correct, try now by opening the app, the first time you will have to do it in the same way as the `pkg` was opened.

> - In the Finder on your Mac, locate the app you want to open.
> - Don’t use Launchpad to do this. Launchpad doesn’t allow you to access the shortcut menu.
> - Control-click the app icon, then choose Open from the shortcut menu.
>
> The app is saved as an exception to your security settings, and you can open it in the future by double-clicking it just as you can any registered app.

![Captura desde 2024-05-02 10-21-28](https://github.com/ollm/OpenComic/assets/18549191/98f1b019-1f7a-41b5-8c69-a0719de41eea)

**Comment by Megs108** [comment]
Hi,

Yes! This version is working. It did immediately show the update for v1.2.0 but I've simply dismissed this for now. 

One thing though; once I opened the application it showed in the dock, launchpad and also on the desktop (I've included the screenshot of the desktop), must I keep this here or can I bin and delete it? I'm just a little apprehensive I'll cause another issue.... sorry. 

![Screenshot 2024-05-02 at 13 28 48](https://github.com/ollm/OpenComic/assets/154489873/b6785735-f435-430d-895b-60e0cdecb943)


Also about v1.2.0 - should I just ignore this for now and only try an update when you have a new one?

**Comment by ollm** [comment]
> One thing though; once I opened the application it showed in the dock, launchpad and also on the desktop (I've included the screenshot of the desktop), must I keep this here or can I bin and delete it? I'm just a little apprehensive I'll cause another issue.... sorry.

You can try deleting it to see what happens, if it is a shortcut nothing should happen.

> Also about v1.2.0 - should I just ignore this for now and only try an update when you have a new one?

I think it will not be necessary, since it seems to me that I have found the cause of the error, I have been checking if any OpenComic dependency is incompatible with macOS Catalina 10.15.7 and `sharp` (Dependency that is used to generate the thumbnails and enhance the downscaled images on reading) had a bug in macOS 10.15 or lower https://github.com/lovell/sharp/issues/3438, I have updated to a fixed version of `sharp`.

If you can try the v1.2.0 with the fix 153902cba8d8d8592919bc234e1cef52a00f4f5b

https://mega.nz/file/yPIFWIDa#ZjcJWB0A8ELGNnOiXIG7mOYFqKXIg2uTQauR9TJcbjE

**Comment by Megs108** [comment]
Hi,

No problems after deleting, thankfully, and this new version of v1.2.0 works! Very happy over here.

Thank you for all your help and patience with explaining things to me.

**Comment by ollm** [comment]
Thanks to you too for trying all the suggestions.

**Comment by Megs108** [comment]
:)
